core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -111,8 +111,11 @@ private[spark] class TaskSetManager(
 
   val weight = 1
   val minShare = 0
-  var priority = taskSet.priority
-  var stageId = taskSet.stageId
+  val priority = if (taskSet.properties != null) {
+    taskSet.properties.getProperty("spark.sql.execution.id", "0").toLong
```

Comment:
Although we has a similar one inside `AppStatusListener` already, this means another implicit dependency to `sql` module.

Reply:
BTW, you may want to define `private val SQL_EXECUTION_ID_KEY = "spark.sql.execution.id"` like `AppStatusListener`. Or, if we need this, we may need to define this at the more higher common place.